File: Personal Projects/Udemy Machine Learning Course/Homework Projects/Part 2 - Regression/Section 4 - Simple Linear Regression/my_simple_linear_regression.py
#!/usr/bin/env python2
# -*- coding: utf-8 -*-
"""
Created on Sun Oct 14 14:21:22 2018

@author: matthew
"""

# Data Preprocessing Template

# Importing the libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Importing the dataset
dataset = pd.read_csv('Salary_Data.csv')

#This code uses imputer to look for missing data and replace NaN values with their column's mean

X = dataset.iloc[:, :-1].values

y = dataset.iloc[:, 1].values



#This code splits the data into the training set and the test set

from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.33, random_state = 42)

#The libraries for simple linear regression automatically handle feature scaling

from sklearn.linear_model import LinearRegression

regressor = LinearRegression()

regressor.fit(X_train, y_train)

y_pred = regressor.predict(X_test)

plt.scatter(X_test, y_test, color='red')
plt.plot(X_train, regressor.predict(X_train), color='blue')
plt.title('Salary vs. Experience (Test Set)')
plt.xlabel('Years of Experience')
plt.ylabel('Salary')
plt.show()

File: Personal Projects/Udemy Machine Learning Course/Homework Projects/Part 2 - Regression/Section 4 - Simple Linear Regression/my_simple_linear_regression.R
dataset = read.csv('Salary_Data.csv')


#Splitting the data into training set and test set
#install.packages('caTools')
library(caTools)

set.seed(143)
split = sample.split(dataset$Salary, SplitRatio = 0.66)
training_set = subset(dataset, split == TRUE)
test_set = subset(dataset, split == FALSE)

#Here is the code for feature scaling

#training_set[, 2:3] = scale(training_set[, 2:3])
#test_set[, 2:3] = scale(test_set[, 2:3])

regressor = lm(formula = Salary ~ YearsExperience,
               training_set)

y_pred = predict(regressor, newdata = test_set)

library(ggplot2)

ggplot() + 
  geom_point(aes(x = training_set$YearsExperience, y = training_set$Salary),
             colour = 'red') +
  geom_line(aes(x = training_set$YearsExperience, y = predict(regressor, newdata = training_set)),
            colour = 'blue') +
  ggtitle('Salary vs. Experience (Training Set)') +
  xlab('Years of Experience') +
  ylab('Salary')
